app/Http/Controllers/App/PostController.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace App\Http\Controllers\App;
+
+use App\Http\Controllers\Controller;
+use App\Models\Banner;
+use App\Models\Challenge;
+use App\Models\Comment;
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+use Inertia\Inertia;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Inertia\Response
+     */
+    public function index(Request $request)
+    {
+        $posts = Post::with(['author', 'team', 'banner', 'challenge', 'comments', 'comments.author', 'users'])->orderByDesc('updated_at')->get();
+        $posts = $posts->each(function($post){$post->team->image = ($post->team->image) ? Storage::disk('upload')->url($post->team->image) : null;});
+
+        $user = $request->user();
+        $teams = $user->teams;
+        $teams->load('banner');
+
+        return Inertia::render('post/Index', [
+            'posts' => $posts,
```

Comment:
We must use a `PostResource` here to prevent leaking the join_code:
![2021-04-28-171410](https://user-images.githubusercontent.com/3864879/116428465-39270380-a845-11eb-9ce0-4e30dd487690.png)

Using the `OtherTeam` will also take care of the `team->image`

Reply:
I had to add lots of Resources to make that work, and had a hard time with the users profile photo. But now it works.